spynnaker/pyNN/models/neuron/synapse_types/synapse_type_exponential.py

```diff
@@ -39,16 +40,18 @@ def data_type(self):
 def get_exponential_decay_and_init(tau, machine_time_step):
     ulfract = pow(2, 32)
     ts = float(machine_time_step) / 1000.0
-    return (
-
-        # decay = e^(-ts / tau) as an unsigned long fract
-        tau.apply_operation(
-            lambda x: int(numpy.exp(-ts / x) * ulfract)),
-
-        # init = (tau / ts) * (1 - e^(-ts / tau)) as an unsigned long fract
-        tau.apply_operation(
-            lambda x: int((x / ts) * (1.0 - numpy.exp(-ts / x)) * ulfract))
-    )
+    # decay = e^(-ts / tau) as an unsigned long fract
+    decay = lambda x: int(numpy.exp(-ts / x) * ulfract)  # noqa E731
```

Comment:
This disables a QA check but I don't believe it should.  It would be better to make decay (and init later) into functions and pass them that way, as the function is used more than once in the code.

See https://stackoverflow.com/questions/45790269/pep-8-warning-do-not-use-a-lambda-expression-use-a-def-for-a-defaultdict-lambd